src/cascadia/inc/cppwinrt_utils.h

```diff
@@ -57,6 +57,53 @@ private:
     winrt::event_token className::name(Windows::Foundation::TypedEventHandler<sender, args> const& handler) { return eventHandler.add(handler); } \
     void className::name(winrt::event_token const& token) noexcept { eventHandler.remove(token); }
 
+// This is a helper macro for both declaring the signature of an event, and
+// defining the body. Winrt events need a method for adding a callback to the
+// event and removing the callback. This macro will both declare the method
+// signatures and define them both for you, because they don't really vary from
+// event to event.
+// Use this in a classes header if you have a Windows.Foundation.TypedEventHandler
+#define TYPED_EVENT(name, sender, args)                                                                                                     \

```

Comment:
is this just a combined version of `DEFINE_..WITH_TYPED` and `DECLARE_..WITH_TYPED`?

Reply:
if so please consider using the decltype magic in #1104 to remove the need for `sender`!

Reply:
You better believe it is. However, I'm a little cautious on doing that as much as I want to. There's some stuff in #2208 that explicitly has a `Object` sender, where the `Object` is not the current type. So for those situations we _explicitly_ need to say `IInspectable` - would the magic still work for that case?